misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/OptionalAssertionsSpec.kt

```diff
@@ -1,43 +1,66 @@
 package ch.tutteli.atrium.specs.integration
 
 import ch.tutteli.atrium.api.fluent.en_GB.messageContains
+import ch.tutteli.atrium.api.fluent.en_GB.toBe
 import ch.tutteli.atrium.api.fluent.en_GB.toThrow
 import ch.tutteli.atrium.api.verbs.internal.expect
+import ch.tutteli.atrium.creating.Expect
 import ch.tutteli.atrium.specs.*
-import ch.tutteli.atrium.translations.DescriptionOptionalAssertion.EMPTY
+import ch.tutteli.atrium.translations.DescriptionOptionalAssertion
 import org.spekframework.spek2.Spek
 import org.spekframework.spek2.style.specification.Suite
 import java.lang.AssertionError
 import java.util.Optional
 
-abstract class OptionalAssertionsSpec (
-    isEmpty: Fun0<Optional<Any>>,
+abstract class OptionalAssertionsSpec(
+    isEmpty: Fun0<Optional<Int>>,
+    isPresentFeature: Feature0<Optional<Int>, Int>,
+    isPresent: Fun1<Optional<Int>, Expect<Int>.() -> Unit>,
     describePrefix: String = "[Atrium] "
 ) : Spek({
 
-    include(object : SubjectLessSpec<Optional<Any>>(
+    include(object : SubjectLessSpec<Optional<Int>>(
         "$describePrefix[Path] ",
-        isEmpty.forSubjectLess()
+        isEmpty.forSubjectLess(),
+        isPresentFeature.forSubjectLess().adjustName { "$it feature" },
+        isPresent.forSubjectLess { toBe(1) }
+    ) {})
+    include(object : AssertionCreatorSpec<Optional<Int>>(
+        describePrefix, Optional.of(2),
+        isPresent.forAssertionCreatorSpec("$toBeDescr: 2") { toBe(2) }
     ) {})
 
     fun describeFun(vararg funName: String, body: Suite.() -> Unit) =
         describeFunTemplate(describePrefix, funName, body = body)
 
-    describeFun(isEmpty.name) {
+    describeFun(isEmpty.name, isPresent.name) {
         val isEmptyFun = isEmpty.lambda
+        val isPresentFun = isPresentFeature.lambda
 
-        context("is empty") {
-            it ("does not throw") {
-                val emptyValue = Optional.empty<Any>()
+        val emptyValue = Optional.empty<Int>()
+        context("$emptyValue") {
+            it("does not throw") {
```

Comment:
```suggestion
            it("${isEmpty.name} - does not throw") {
```
(note, you can also apply suggestions directly from the GitHub website)